### PR TITLE
Adapt HYVideo1.5 accuracy test for npu

### PR DESCRIPTION
**Pull request by BLANKETusers**
## Purpose

- Adapted precision testing for HunyuanVideo-1.5 on NPU and integrated it into the nightly pipeline.

- (CLI override fail)Fixed an issue where HunyuanVideo-1.5 T2V/I2V models ran on a single GPU despite --tensor-parallel-size 2 --vae-patch-parallel-size 2 arguments, causing OOM errors on NPU dual-card setups.

## Test Plan

```
pytest -s -v tests/e2e/accuracy/hunyuanvideo15_t2v/test_hunyuanvideo15_t2v_video_similarity.py
pytest -s -v tests/e2e/accuracy/hunyuanvideo15_i2v/test_hunyuanvideo15_i2v_video_similarity.py
```
**vLLM Version:**
`0.26.0`
**vLLM-Omni Commit:**
`e6782a9f8b9c40bf9230cbdba5cc4e87a5edca44`
## Test Result

i2v

```
============================================================================================================ warnings summary =============================================================================================================
../../../../usr/local/python3.12.13/lib/python3.12/site-packages/torch/jit/_script.py:362: 14 warnings
  /usr/local/python3.12.13/lib/python3.12/site-packages/torch/jit/_script.py:362: DeprecationWarning: `torch.jit.script_method` is deprecated. Please switch to `torch.compile` or `torch.export`.
    warnings.warn(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
--- Running Summary
=============================================================================================== 3 passed, 14 warnings in 399.39s (0:06:39) ================================================================================================
```

t2v

```
============================================================================================================ warnings summary =============================================================================================================
../../../../usr/local/python3.12.13/lib/python3.12/site-packages/torch/jit/_script.py:362: 14 warnings
  /usr/local/python3.12.13/lib/python3.12/site-packages/torch/jit/_script.py:362: DeprecationWarning: `torch.jit.script_method` is deprecated. Please switch to `torch.compile` or `torch.export`.
    warnings.warn(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
--- Running Summary
=============================================================================================== 3 passed, 14 warnings in 358.48s (0:05:58) ================================================================================================
```

### CODE REVIEW

File: .buildkite/npu/test-npu-nightly.yml [hunk 1]
```diff
@@ -185,6 +185,24 @@ steps:
           HF_TOKEN: "${HF_TOKEN}"
         commands:
           - pytest -s -v tests/e2e/online_serving/test_wan22_expansion.py -k npu_i2v --run-level "full_model"
+      - label: ":full_moon: HunyuanVideo-1.5 T2V · Accuracy Test"
```
Comment (yenuo26) [inline .buildkite/npu/test-npu-nightly.yml:188]: Can you move these two jobs to the 'Diffusion X2V model Test' group?

File: tests/e2e/accuracy/hunyuanvideo15_i2v/test_hunyuanvideo15_i2v_video_similarity.py [hunk 2]
```diff
@@ -180,12 +194,12 @@ def _generate_online_video(
...
 
 @pytest.mark.benchmark
-@hardware_test(res={"cuda": "H100"}, num_cards=1)
+@hardware_test(res={"cuda": "H100", "npu": "A3"}, num_cards={"cuda": 1, "npu": 2})
 def test_hunyuanvideo15_i2v_diffusers_offline_generates_video(
     hunyuanvideo15_i2v_image_source: str | None,
 ) -> None:
-    if not torch.cuda.is_available():
-        pytest.skip("HunyuanVideo-1.5 I2V offline accuracy test requires CUDA.")
+    if not is_accelerator_available():
```
Comment (yenuo26) [inline tests/e2e/accuracy/hunyuanvideo15_i2v/test_hunyuanvideo15_i2v_video_similarity.py]: Why do we need this check? It would be better to just use a mark/label to indicate the runner type.

File: vllm_omni/config/config_factory.py [hunk 3]
```diff
@@ -603,8 +603,22 @@ def create_default_diffusion(cls, kwargs: dict[str, Any]) -> list[dict[str, Any]
...
-                devices += f",{i}"
+            # Use actual device IDs from environment if available
+            import os
+
+            visible_devices_env = os.environ.get("ASCEND_RT_VISIBLE_DEVICES") or os.environ.get("CUDA_VISIBLE_DEVICES")
+            if visible_devices_env and num_devices > 1:
+                # Parse comma-separated device IDs
+                available_device_ids = [d.strip() for d in visible_devices_env.split(",") if d.strip()]
+                if len(available_device_ids) >= num_devices:
+                    devices = ",".join(available_device_ids[:num_devices])
```
Comment (lishunyang12) [inline vllm_omni/config/config_factory.py:614]: Use logical device indices here; the runtime already maps them to physical devices.

File: .buildkite/npu/test-npu-nightly.yml [hunk 4]
```diff
@@ -241,6 +241,24 @@ steps:
...
+        timeout_in_minutes: 180
+        mirror_hardwares: a3_npu_2
+        env:
+          VLLM_WORKER_MULTIPROC_METHOD: spawn
+          DIFFUSION_ATTENTION_BACKEND: TORCH_SDPA
+          HF_TOKEN: "${HF_TOKEN}"
+        commands:
+          - pytest -s -v tests/e2e/accuracy/hunyuanvideo15_t2v/test_hunyuanvideo15_t2v_video_similarity.py::test_hunyuanvideo15_t2v_diffusers_offline_generates_video -m "full_model and A3" --run-level full_model
```
Comment (lishunyang12) [inline .buildkite/npu/test-npu-nightly.yml:252]: Run the online and similarity tests too; this job only checks video metadata.
